app/js/stores/AnswerStore.js

```diff
@@ -1,114 +1,88 @@
-
-    case ZhishiConstants.QUESTION_UPDATE:
-      if (action.data) {
-        loadAnswers(action.data.id, action.data.answers)
-        AnswerStore.emitChange();
-      }
-      break;
-
-    case ZhishiConstants.ANSWER_INDEX:
-      if (action.data) {
-        loadAnswers(action.data.question_id, action.data.answers)
-        AnswerStore.emitChange();
-      }
-      break;
-
-    // case ZhishiConstants.QUESTION_UPDATE:
-    //   if (action.data && action.data.answers) {
-    //     loadAnswers(action.data.id, action.data.answers)
-    //   }
-    //   break;
-
-    case ZhishiConstants.ANSWER_EDIT:
-      if (action.data) {
-        edit(action.data.question_id, action.data.id)
-        AnswerStore.emitChange();
-      }
-      break;
 
-    case ZhishiConstants.ANSWER_UPDATE:
-      if (action.data) {
-        update(action.data);
-        AnswerStore.emitChange();
-      }
-      break;
+  edit(data) {
+    this._answers[data.question_id][data.id]['status'] = 'editing editor-content';
+  }
 
-    case ZhishiConstants.ANSWER_UPDATE_VOTES:
-      if (action.data) {
-        update_votes_count({...action.data})
-        AnswerStore.emitChange();
-      }
-      break;
+  update(answer) {
+    Common.update(this._answers[answer.question_id], answer.id, answer);
+  }
+  getAnswer(question_id, id) {
+    return this._answers[question_id][id];
+  }
 
-    default:
-      // nothing for now
+  getAnswers(question_id) {
+    return this._answers[question_id];
   }
-});
+  getTopAnswers() {
+    return this._top_answers;
+  }
+  update_votes_count(data) {
+    this._answers[data.meta.question_id][data.id]['votes_count'] = data.votes_count.response;
+    this._answers[data.meta.question_id][data.id]['user_vote'] = data.value;
+  }
+  _registerActions(action) {
+    switch (action.actionType) {
+
+      case ZhishiConstants.QUESTION_UPDATE:
+        if (action.data) {
+          this.loadAnswers(action.data.id, action.data.answers);
+          this.emitChange();
+        }
+        break;
+
+      case ZhishiConstants.ANSWER_INDEX:
+        if (action.data) {
+          this.loadAnswers(action.data.question_id, action.data.answers);
+          this.emitChange();
+        }
+        break;
+
+      // case ZhishiConstants.QUESTION_UPDATE:
+      //   if (action.data && action.data.answers) {
+      //     loadAnswers(action.data.id, action.data.answers)
+      //   }
+      //   break;
+
+      case ZhishiConstants.ANSWER_EDIT:
+        if (action.data) {
+          this.edit(action.data);
+          this.emitChange();
+        }
+        break;
+
+      case ZhishiConstants.ANSWER_UPDATE:
+        if (action.data) {
+          this.update(action.data);
+          this.emitChange();
+        }
+        break;
+
+      case ZhishiConstants.ANSWER_UPDATE_VOTES:
+        if (action.data) {
+          this.update_votes_count(action.data);
+          this.emitChange();
+        }
+        break;
+
+      default:
+        // nothing for now
+    }
+  }
+}
```

Comment:
Yes. The do never get exported. But not having them enclosed in a particular class or constant or object is just as bad.
There is no private method implementation in es6 yet. The best we could do is to prefix it with an underscore _. the plus side is it doesn't let us repeat logic.